File: Day/241024/회전초밥_2531/윤병희.swift
let input = readLine()!.split(separator: " ").map{ Int($0)! }
var array = [Int]()
var getArray = Array(repeating: 0, count: input[1]+1)

for _ in 1...input[0] {
    array.append(Int(readLine()!)!)
}

array += array[0...input[2]-1]

var nowMax = 0
var start = 0
var end = input[2]-1

for i in 0...input[2] - 1{
    if getArray[array[i]] == 0 {
        nowMax += 1
    }

    getArray[array[i]] += 1
}

var resultMax = nowMax

while start < input[0] {
    start += 1
    end += 1

    if getArray[array[start-1]] == 1 {
        nowMax -= 1
    }
    getArray[array[start-1]] -= 1

    if getArray[array[end]] == 0 {
        nowMax += 1
    }
    getArray[array[end]] += 1
    
    if getArray[input[3]] == 0 {
        resultMax = max(resultMax, nowMax + 1)
    } else {
        resultMax = max(resultMax, nowMax)
    }
}

print(resultMax)

File: Day/250103/1의개수세기_9527/윤병희.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        long A = Long.parseLong(input[0]);
        long B = Long.parseLong(input[1]);

        System.out.println(countOnes(B) - countOnes(A - 1));
    }

    private static long countOnes(long x) {
        if (x < 0) return 0;

        long count = 0;
        long bitPosition = 1;

        while (bitPosition <= x) {
            long completeGroups = (x + 1) / (bitPosition * 2);
            long remainder = (x + 1) % (bitPosition * 2);

            count += completeGroups * bitPosition;
            count += Math.max(0, remainder - bitPosition);

            bitPosition *= 2;
        }

        return count;
    }
}
